ARDemos/ARDemos/Extensions.swift

```diff
@@ -0,0 +1,23 @@
+import SceneKit
+
+extension float4x4 {
+    var translation: float3 {
+        let translation = self.columns.3
+        return float3(translation.x, translation.y, translation.z)
+    }
+}
+
+protocol Loopable {
+    var allProperties: [String: Any] { get }
+}
+extension Loopable {
+    var allProperties: [String: Any] {
+        var result = [String: Any]()
+        Mirror(reflecting: self).children.forEach { child in
```

Comment:
This is an interesting use of Mirror, would you like to explain a little bit more why you are using it?